### PR TITLE
feat: add permissions to User change and delete mutations

### PR DESCRIPTION
**Pull request by cacarrara**
This change adds the permissions check to change and delete User
mutations. Only the user itself can perform these operations.

This is part of:
- #75

### CODE REVIEW

File: terraso_backend/apps/graphql/schema/users.py [hunk 1]
```diff
@@ -54,8 +55,12 @@ class Input:
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **kwargs):
+        request_user = info.context.user
         _id = kwargs.pop("id")
 
+        if str(request_user.id) != _id:
+            raise GraphQLValidationException("User has no permission to change User.")
```
Comment (paulschreiber) [inline terraso_backend/apps/graphql/schema/users.py:62]: issue: This messages is unclear. Perhaps indicate they cannot change a different user/someone else/a user other than themselves.

File: terraso_backend/apps/graphql/schema/users.py [hunk 2]
```diff
@@ -76,3 +81,13 @@ class UserDeleteMutation(BaseDeleteMutation):
...
     class Input:
         id = graphene.ID()
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        request_user = info.context.user
+        _id = kwargs.get("id")
+
+        if str(request_user.id) != _id:
+            raise GraphQLValidationException("User has no permission to delete User.")
```
Comment (paulschreiber) [inline terraso_backend/apps/graphql/schema/users.py:91]: issue: Same as above.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

**Comment by cacarrara** [comment]
@paulschreiber messages were updated. Let me know if it's better now.

File: terraso_backend/apps/graphql/schema/users.py [hunk 3]
```diff
@@ -76,3 +83,15 @@ class UserDeleteMutation(BaseDeleteMutation):
...
         id = graphene.ID()
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        request_user = info.context.user
+        _id = kwargs.get("id")
+
+        if str(request_user.id) != _id:
+            raise GraphQLValidationException(
+                "User has no permission to delete another User information."
```
Comment (paulschreiber) [inline terraso_backend/apps/graphql/schema/users.py:94]:
```suggestion
                f"User {request_user.id} cannot delete information for user {_id}"
```

File: terraso_backend/apps/graphql/schema/users.py [hunk 4]
```diff
@@ -54,8 +55,14 @@ class Input:
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **kwargs):
+        request_user = info.context.user
         _id = kwargs.pop("id")
 
+        if str(request_user.id) != _id:
+            raise GraphQLValidationException(
+                "User has no permission to change another User information."
```
Comment (paulschreiber) [inline terraso_backend/apps/graphql/schema/users.py:63]:
```suggestion
                f"User {request_user.id} cannot change information for user {_id}"

```

**Review by paulschreiber — CHANGES_REQUESTED** [review]

**Comment by cacarrara** [comment]
It's done @paulschreiber

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**